### PR TITLE
[GOBBLIN-1325] Add Kafka 1.1 Module

### PR DESCRIPTION
**Pull request by hanghangliu**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1325

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by shirshanka** [comment]
@hanghangliu : I have a suggestion to make this be just kafka-1  (instead of separating 1.1 and future 1.x modules)  ... we will just upgrade the version of 1.x over time. 

So gobblin-kafka-11/build.gradle  -> gobblin-kafka-1/build.gradle 

and Kafka11DataWriter -> Kafka1DataWriter

We had to separate 08 and 09 in the past because there were backwards incompatible changes that Kafka made ... hopefully that is not true anymore.

File: gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/client/Kafka11ConsumerClient.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    @EqualsAndHashCode(callSuper = true)
+    @ToString
+    public static class Kafka11ConsumerRecord<K, V> extends BaseKafkaConsumerRecord implements
+            DecodeableKafkaRecord<K, V> {
+        private final ConsumerRecord<K, V> consumerRecord;
+
+        public Kafka11ConsumerRecord(ConsumerRecord<K, V> consumerRecord) {
+            // Kafka 09 consumerRecords do not provide value size.
+            // Only 08 and 11 versions provide them.
+            super(consumerRecord.offset(), BaseKafkaConsumerRecord.VALUE_SIZE_UNAVAILABLE, consumerRecord.topic(), consumerRecord.partition());
```
Comment (shirshanka) [inline gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/client/Kafka11ConsumerClient.java:208]: Provide value size here?

File: gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/client/Kafka11ConsumerClient.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+
+/**
+ * A {@link GobblinKafkaConsumerClient} that uses kafka 1.1 consumer client. Use {@link Factory#create(Config)} to create
+ * new Kafka1.1ConsumerClients. The {@link Config} used to create clients must have required key {@value #GOBBLIN_CONFIG_VALUE_DESERIALIZER_CLASS_KEY}
+ *
+ * @param <K> Message key type
+ * @param <V> Message value type
+ */
+public class Kafka11ConsumerClient<K, V> extends AbstractBaseKafkaConsumerClient {
```
Comment (sv2000) [inline gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/client/Kafka11ConsumerClient.java:61]: Can we add implementations of subscribe() methods similar to what is done for Kafka09ConsumerClient?

**Review by sv2000 — CHANGES_REQUESTED** [review]

File: gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/client/Kafka11ConsumerClient.java (hunk 2 above)
Comment (sv2000) [reply gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/client/Kafka11ConsumerClient.java:61]: Similarly, add implementations for getMetrics() and commitOffsets() methods.

File: gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/serialize/LiAvroSerializer.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Serializer;
+
+
+/**
+ * LinkedIn's implementation of Avro-schema based serialization for Kafka
+ * TODO: Implement this for IndexedRecord not just GenericRecord
+ *
+ */
+public class LiAvroSerializer extends LiAvroSerializerBase implements Serializer<GenericRecord> {
```
[the remark this answers is not in the document]
Comment (hanghangliu) [reply gobblin-modules/gobblin-kafka-11/src/main/java/org/apache/gobblin/kafka/serialize/LiAvroSerializer.java]: It's a good idea to deprecate the LI-specific implementations. I'm just using them during tests. 

In the meantime, could you please tell me what avro serde do we normally use? I couldn't find a better implementation in our codebase.

**Comment by hanghangliu** [comment]
> @hanghangliu : I have a suggestion to make this be just kafka-1 (instead of separating 1.1 and future 1.x modules) ... we will just upgrade the version of 1.x over time.
> 
> So gobblin-kafka-11/build.gradle -> gobblin-kafka-1/build.gradle
> 
> and Kafka11DataWriter -> Kafka1DataWriter
> 
> We had to separate 08 and 09 in the past because there were backwards incompatible changes that Kafka made ... hopefully that is not true anymore.

Good idea. I have changed the module as kafka-1.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request closed without merging**